crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -0,0 +1,948 @@
+            res_type: r_type,
+            res_name: String::new(),
+            is_deleted: false,
+            parent_block: p_block,
+            bit_size: 0,
+            max_value: BigUint::zero(),
+            phi_arguments: Vec::new(),
+        }
+    }
+
+    pub fn print_i(&self) -> String {
+        if self.res_name.is_empty() {
+            format!("{:?}", self.idx)
+        } else {
+            self.res_name.clone()
+        }
+    }
+
+    //indicates if the operation is a substraction
+    pub fn is_sub(&self) -> bool {
+        matches!(self.operator, Operation::sub | Operation::safe_sub)
+    }
+
+    //indicates whether the left and/or right operand of the instruction is required to be truncated to its bit-width
+    pub fn truncate_required(&self, lhs_bits: u32, rhs_bits: u32) -> (bool, bool) {
+        match self.operator {
+            Operation::add => (false, false),
+            Operation::safe_add => (false, false),
+            Operation::sub => (false, false),
+            Operation::safe_sub => (false, false),
+            Operation::mul => (false, false),
+            Operation::safe_mul => (false, false),
+            Operation::udiv => (true, true),
+            Operation::sdiv => (true, true),
+            Operation::urem => (true, true),
+            Operation::srem => (true, true),
+            Operation::fmod => (true, true),   //to check
+            Operation::fneg => (false, false), //to check
+            Operation::fdiv => (true, true),   //to check
+            Operation::div => (false, false),
+            Operation::eq => (true, true),
+            Operation::ne => (true, true),
+            Operation::ugt => (true, true),
+            Operation::uge => (true, true),
+            Operation::ult => (true, true),
+            Operation::ule => (true, true),
+            Operation::sgt => (true, true),
+            Operation::sge => (true, true),
+            Operation::slt => (true, true),
+            Operation::sle => (true, true),
+            Operation::lt => (true, true),
+            Operation::gt => (true, true),
+            Operation::lte => (true, true),
+            Operation::gte => (true, true),
+            Operation::and => (true, true),
+            Operation::not => (true, true),
+            Operation::or => (true, true),
+            Operation::xor => (true, true),
+            Operation::cast => {
+                if lhs_bits != rhs_bits {
+                    return (false, true);
+                }
+                (false, false)
+            }
+            Operation::ass => {
+                assert!(lhs_bits == rhs_bits);
+                (false, false)
+            }
+            Operation::trunc | Operation::phi => (false, false),
+            Operation::nop | Operation::jne | Operation::jeq | Operation::jmp => (false, false),
+            Operation::eq_gate => (true, true),
+        }
+    }
+
+    pub fn get_max_value(&self, lhs_max: BigUint, rhs_max: BigUint) -> BigUint {
```

Comment:
I don't think this belongs here in this file. Can we add a comment to it specifying its usecase, as we have `Field` and `BigUint`